recipes/icon/build.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/env bash
+
+# Abort script upon untested failure
+set -e
+
+make Configure name=linux
+
+make CC=${CC}
+
+# Run a small sample of the test suite
+make Samples
+
+# Deploy to icon and bin directories at PREFIX
+mkdir -p ${PREFIX}/bin
+DEPLOY=${PREFIX}/icon
+mkdir -p ${DEPLOY}
+
+# remove compilation intermediates
+make Clean || echo Clean exit code is $?
+
+# Install by copy to give users full access to source, e.g., so that they can
+#   use it for reference when building loadable C functions.
+cp -R ${SRC_DIR}/* ${DEPLOY}
```

Comment:
I don't understand why you need the source

Reply:
@isuruf You cannot build loadable C extensions without the source including the Makedefs

Reply:
Looks like there's only a few in https://www2.cs.arizona.edu/icon/library/ccfuncs.htm
Why can't they be precompiled and packaged in the conda package?

Reply:
@isuruf Thank you, but I am afraid that you have misunderstood. The point is not to provide what has already been done; indeed, the extensions that you referenced are indeed precompiled as part of the build and test.  The point is to provide for people to build extensions as needed (think analogously of loading an R package, which invokes compilers in the background).

For an example of what the *user* should be able to build, see:
https://github.com/gtownsend/icon/tree/master/ipl/packs/loadfunc

As a "compromise" (or capitulation), I can strip it down to the point that the user can run a script to get the missing pieces including `sed`, `make`, `binutils` that they need to build their external functions.  It may be a *lot* of work to tear apart what works nicely when it's all together and wriite a script to reassemble it it all again - that seems counter to the point of conda packaging everything that you need.

I really need to get this into conda-forge so that I can use it in Galaxy, even if I have to settle for only a partially functional package.

Reply:
@isuruf Here's another problem: I cannot find a link to download the IPL *in toto* from the official Icon pages at https://cs.arizona.edu/icon; if you find a link, I'd love to know about it. So, it does make sense to include the IPL with the package.  It isn't very big, especially if I prune out the graphics procedures.

Reply:
From a brief look only `mklib.sh`, `Makedefs` and `icall.h` is needed right? I dont see why we need to distribute the whole source when only those 3 files are needed. Can you show an example of a loadable extension that is not inside icon source?

Reply:
@isruf Please see the note that the IPL is not available otherwise.  The IPL is *not* the source used to build Icon.  I'll push something this afternoon.  `du -sh` reports only 14 Mb. No source except the IPL and essentially the three files that you mentioned.  This is tested.

```bash
make Install dest=${DEPLOY}

cp ${SRC_DIR}/Makedefs ${DEPLOY}/Makedefs
mkdir -p ${DEPLOY}/ipl/cfuncs; cp ${SRC_DIR}/ipl/cfuncs/*   ${DEPLOY}/ipl/cfuncs/
mkdir -p ${DEPLOY}/ipl/data  ; cp ${SRC_DIR}/ipl/data/*     ${DEPLOY}/ipl/data/
mkdir -p ${DEPLOY}/ipl/docs  ; cp ${SRC_DIR}/ipl/docs/*     ${DEPLOY}/ipl/docs/
mkdir -p ${DEPLOY}/ipl/incl  ; cp ${SRC_DIR}/ipl/incl/*     ${DEPLOY}/ipl/incl/
mkdir -p ${DEPLOY}/ipl/packs ; cp -R ${SRC_DIR}/ipl/packs/* ${DEPLOY}/ipl/packs
mkdir -p ${DEPLOY}/ipl/procs ; cp ${SRC_DIR}/ipl/procs/*    ${DEPLOY}/ipl/procs/
mkdir -p ${DEPLOY}/ipl/progs ; cp ${SRC_DIR}/ipl/progs/*    ${DEPLOY}/ipl/progs/

# Create instructions to make extension libraries in C
mkdir -p ${DEPLOY}/doc ; cp ${SRC_DIR}/doc/* ${DEPLOY}/doc/
echo '
To build loadable C functions as described at
  https://www.cs.arizona.edu/icon/uguide/cfuncs.htm
you will need to add the C compiler and some tools to your environment, e.g.:
  conda install -c conda-forge gcc_linux-64 make sed binutils
  pushd ${CONDA_PREFIX}/icon/ipl/packs/loadfunc
  make CC=${CC}
  ls -l libdemo.so
  popd

Bug: Building with C++, e.g.,
  ${CONDA_PREFIX}/icon/ipl/packs/loadfuncpp
has not yet been successful in a conda environment.
It will be necessary but not sufficient to install gxx_linux-64.
' >> ${DEPLOY}/doc/condagcc.txt

```

Reply:
My latest commit 209d33c reflects this.
```bash
$ conda activate icon

$ cd $CONDA_PREFIX

$ du -sh --apparent-size icon/* *
571     icon/Makedefs
1.3K    icon/README
418K    icon/bin
127K    icon/doc
4.2M    icon/ipl
4.6M    icon/lib
15K     icon/man
690     README_icon
4.1K    bin
571K    conda-meta
18K     etc
4.0K    icon

$ du -sh --apparent-size icon
9.4M    icon
```